File: apps/ios/Velix/Features/Dashboard/ServerRow.swift
import SwiftUI

struct ServerRow: View {
    let server: ServerSummary

    var body: some View {
        HStack(spacing: 14) {
            ZStack {
                Circle()
                    .fill(statusColor.opacity(0.15))
                    .frame(width: 38, height: 38)
                Image(systemName: "server.rack")
                    .font(.system(size: 15, weight: .semibold))
                    .foregroundStyle(statusColor)
            }

            VStack(alignment: .leading, spacing: 3) {
                Text(server.name)
                    .font(.system(size: 16, weight: .semibold))
                if let metricsText {
                    Text(metricsText)
                        .font(.system(size: 13))
                        .foregroundStyle(.secondary)
                }
            }

            Spacer()

            Circle()
                .fill(statusColor)
                .frame(width: 8, height: 8)
        }
        .padding(.vertical, 4)
    }

    private var statusColor: Color {
        switch server.status {
        case "ONLINE": return .green
        case "ERROR": return .red
        default: return .gray // OFFLINE, PENDING, e qualquer status desconhecido
        }
    }

    private var metricsText: String? {
        guard let metrics = server.metrics else { return nil }
        var parts: [String] = []
        if let cpu = metrics.cpuPercent {
            parts.append("CPU \(Int(cpu.rounded()))%")
        }
        if let usedMb = metrics.memUsedMb, let totalMb = metrics.memTotalMb, totalMb > 0 {
            let memPercent = Int((Double(usedMb) / Double(totalMb) * 100).rounded())
            parts.append("Mem \(memPercent)%")
        }
        if let temp = metrics.temperatureCelsius {
            parts.append("\(Int(temp.rounded()))°C")
        }
        return parts.isEmpty ? nil : parts.joined(separator: " · ")
    }
}

#Preview {
    List {
        ServerRow(server: ServerSummary(
            id: "1", name: "prod-01", status: "ONLINE", publicIp: nil, hostname: nil,
            dockerInstalled: true,
            metrics: ServerMetrics(loadAvg: nil, memUsedMb: 2048, memTotalMb: 4096, diskPercent: nil, cpuPercent: 42.3, temperatureCelsius: 55.1)
        ))
        ServerRow(server: ServerSummary(
            id: "2", name: "staging", status: "PENDING", publicIp: nil, hostname: nil,
            dockerInstalled: false, metrics: nil
        ))
        ServerRow(server: ServerSummary(
            id: "3", name: "backup", status: "ERROR", publicIp: nil, hostname: nil,
            dockerInstalled: true,
            metrics: ServerMetrics(loadAvg: nil, memUsedMb: nil, memTotalMb: nil, diskPercent: nil, cpuPercent: 91.0, temperatureCelsius: nil)
        ))
    }
}

File: apps/android/app/src/main/java/com/velix/app/features/dashboard/ServerRow.kt
package com.velix.app.features.dashboard

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import com.velix.app.core.ServerMetrics
import com.velix.app.core.ServerSummary
import kotlin.math.roundToInt

/** Uma linha da lista do dashboard: indicador de status + nome + métricas
 * (quando presentes). Espelha o `ServerRow` do app iOS — mesma lógica de
 * cores e de formatação de CPU/mem/temperatura. */
@Composable
fun ServerRow(server: ServerSummary, onClick: () -> Unit) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .clickable(onClick = onClick)
            .padding(horizontal = 24.dp, vertical = 12.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(14.dp),
    ) {
        Box(
            modifier = Modifier
                .size(38.dp)
                .background(color = statusColor(server.status).copy(alpha = 0.15f), shape = CircleShape),
            contentAlignment = Alignment.Center,
        ) {
            Text(
                server.name.take(1).uppercase(),
                style = MaterialTheme.typography.titleMedium,
                color = statusColor(server.status),
            )
        }
        Column(modifier = Modifier.weight(1f)) {
            Text(server.name, style = MaterialTheme.typography.bodyLarge)
            metricsText(server.metrics)?.let {
                Text(it, style = MaterialTheme.typography.bodySmall, color = MaterialTheme.colorScheme.onSurfaceVariant)
            }
        }
        Box(
            modifier = Modifier
                .size(8.dp)
                .background(color = statusColor(server.status), shape = CircleShape),
        )
    }
}

private fun statusColor(status: String): Color = when (status) {
    "ONLINE" -> Color(0xFF4CAF50)
    "ERROR" -> Color(0xFFF44336)
    else -> Color(0xFF9E9E9E) // OFFLINE, PENDING, e qualquer status desconhecido
}

/** Junta CPU/mem/temperatura num texto único, omitindo qualquer campo `null`
 * (nunca mostra "N/A" ou 0 falso). `null` se `metrics` for `null` ou não
 * sobrar nenhum campo pra mostrar. */
private fun metricsText(metrics: ServerMetrics?): String? {
    if (metrics == null) return null
    val parts = mutableListOf<String>()
    metrics.cpuPercent?.let { parts += "CPU ${it.roundToInt()}%" }
    val usedMb = metrics.memUsedMb
    val totalMb = metrics.memTotalMb
    if (usedMb != null && totalMb != null && totalMb > 0) {
        val memPercent = (usedMb.toDouble() / totalMb * 100).roundToInt()
        parts += "Mem $memPercent%"
    }
    metrics.temperatureCelsius?.let { parts += "${it.roundToInt()}°C" }
    return parts.takeIf { it.isNotEmpty() }?.joinToString(" · ")
}
